builder/chroots.go

```diff
@@ -0,0 +1,637 @@
+		version = b.Mixver
+	} else {
+		fmt.Printf("Creating chroots for version %s\n", b.Clearver)
+		version = b.Clearver
+	}
+	// TODO: This validation should happen when reading the configuration.
+	if version == "" {
+		return errors.Errorf("no Mixver or Clearver set, unable to proceed")
+	}
+
+	// TODO: Check if this is redundant with the same cleanup in builder.go!
+	chrootVersionDir := filepath.Join(chrootDir, version)
+	fmt.Printf("Removing pre-existing %s before starting\n", chrootVersionDir)
+	err = os.RemoveAll(chrootVersionDir)
+	if err != nil {
+		return err
+	}
+
+	fmt.Printf("Preparing new %s\n", chrootVersionDir)
+	// TODO: Fix this output when we take bundles from multiple places.
+	fmt.Printf("  based on bundles from: %s\n", b.Bundledir)
+	fmt.Printf("  and yum config: %s\n", b.Yumconf)
+
+	err = os.MkdirAll(chrootVersionDir, 0755)
+	if err != nil {
+		return err
+	}
+	for name, bundle := range set.Bundles {
+		// TODO: Should we embed this information in groups.ini? (Maybe rename it to bundles.ini)
+		var includes bytes.Buffer
+		for _, inc := range bundle.DirectIncludes {
+			fmt.Fprintf(&includes, "%s\n", inc)
+		}
+		err = ioutil.WriteFile(filepath.Join(chrootVersionDir, name+"-includes"), includes.Bytes(), 0644)
+		if err != nil {
+			return err
+		}
+	}
+
+	fmt.Println("Creating os-core chroot")
+	osCoreDir := filepath.Join(chrootVersionDir, "os-core")
+	err = os.MkdirAll(filepath.Join(osCoreDir, "var/lib/rpm"), 0755)
+	if err != nil {
+		return err
+	}
+
+	fmt.Println("Initializing RPM database")
+	_, err = run(
+		"rpm",
+		"--root", osCoreDir,
+		"--initdb",
+	)
+	if err != nil {
+		return err
+	}
+
+	yumCmd := []string{
+		"yum",
+		"--config=" + b.Yumconf,
+		"-y",
+		"--releasever=" + b.Clearver,
+	}
+	// When in Fedora, call dnf instead of yum.
+	if osInfo, oerr := readOSInfo(); oerr == nil {
+		if osInfo.ID == "fedora" {
+			// TODO: Simplify this when we can assume all Fedora users will be >= 22?
+			versionID, _ := strconv.Atoi(osInfo.VersionID)
+			if versionID >= 22 {
+				yumCmd[0] = "dnf"
+			}
+		}
+	}
+
+	fmt.Println("Installing filesystem package in os-core")
+	_, err = run(merge(yumCmd,
+		"--installroot="+osCoreDir,
+		"install",
+		"filesystem",
+	)...)
+	if err != nil {
+		return err
+	}
+
+	clearDir := filepath.Join(osCoreDir, "usr/share/clear")
+	err = os.MkdirAll(filepath.Join(clearDir, "bundles"), 0755)
+	if err != nil {
+		return err
+	}
+
+	fmt.Println("Installing packages in os-core")
+	err = installPackagesToBundleChroot(yumCmd, chrootVersionDir, set.Bundles["os-core"])
+	if err != nil {
+		return err
+	}
+
+	// Writing special files identifying the version in os-core.
+	err = ioutil.WriteFile(filepath.Join(clearDir, "version"), []byte(version), 0644)
+	if err != nil {
+		return err
+	}
+	// TODO: This seems to be the only thing that makes two consecutive chroots of the same
+	// version to be different. Use SOURCE_DATE_EPOCH if available?
```

Comment:
SOURCE_DATE_EPOCH would be a good change as it is used in Clear Linux package builds. This was also discussed as a change for swupd-server.